src/qforte/utils/moment_energy_corrections.py

```diff
@@ -0,0 +1,99 @@
+import qforte as qf
+from qforte.utils.trotterization import trotterize
+from qforte.utils.point_groups import sq_op_find_symmetry
+import numpy as np
+
+def construct_moment_space(self):
+
+    # List that will hold the orbital energy differences used in Moller-Plesset denominators
+    self._mpdenom = []
+    # List that will hold <Phi_k| H |Phi_k> used in the Epstein-Nesbet denominators
+    self._epstein_nesbet = []
+
+    self._E_mmcc_mp = []
+    self._E_mmcc_en = []
+
+    # List that will store the integers representing the determinants used in the moment corrections
+    self._mmcc_excitation_indices = []
+    self._mmcc_pool = qf.SQOpPool()
+
+    if not hasattr(self, '_orb_e'):
+        self.build_orb_energies()
+
+    # create a pool of particle number, Sz, and spatial symmetry adapted second quantized operators
+    # Encode the occupation list into a bitstring
+    ref = sum([b << i for i, b in enumerate(self._ref)])
+    # `& mask_alpha` gives the alpha component of a bitstring. `& mask_beta` does likewise.
+    mask_alpha = 0x5555555555555555
+    mask_beta = mask_alpha << 1
+    nalpha = sum(self._ref[0::2])
+    nbeta = sum(self._ref[1::2])
+
+    if not isinstance(self._max_moment_rank, int) or self._max_moment_rank < 0:
+        raise TypeError("The maximum moment rank must be a non-negative integer!")
+    elif self._max_moment_rank > nalpha + nbeta:
+        # WARNING: This step could be potetnially removed, if QForte allows for methods
+        #          that violate particle number symmetry
+        self._max_moment_rank = nalpha + nbeta
+
+    for I in range(1 << self._nqb):
```

Comment:
Can we factor out this symmetry selection code? I remember similar code elsewhere in this PR push.

Reply:
There are some minor differences between the two codes, but we should be able to factor it out.

Reply:
Agreed privately, that this code isn't stable enough to factor out yet.